### ISSUE TITLE
model with lstmcell gives different results before and after tflite conversion

### ISSUE THREAD

**Issue by DerryFitz** [tensorflow/tensorflow#58858]
### 1. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): OpenSuse TumbleWeed 
- TensorFlow installation (pip package or built from source): pip
- TensorFlow library (version, if pip package or github SHA, if built from source): 2.10

### 2. Code

When I attempt to convert a working tensorflow model to tflite, the model converts, but the results obtained are different
See code below:
```
import tensorflow as tf
from tensorflow.keras.layers import (
    Concatenate,
    Conv2D,
    Conv2DTranspose,
    Dropout,
    BatchNormalization)
import numpy as np

inp = tf.keras.Input([1,192], batch_size = 1)
state_h = tf.keras.Input([192], batch_size = 1)
state_c = tf.keras.Input([192], batch_size = 1)
num_channels = 24

xT = BatchNormalization()(inp)

lstm_in = tf.keras.activations.tanh(xT)
x, new_states = tf.keras.layers.LSTMCell(xT.shape[2])(lstm_in[:,0,:], states = [state_h, state_c])
new_state_h = new_states[0]
new_state_c = new_states[1]
out = x + xT

my_model = tf.keras.Model(inputs=[inp, state_h, state_c],outputs = [ out, 
                                                                       new_state_h,
                                                                       new_state_c])

#save model and create tflite model
my_model.save('lstm_test', include_optimizer = False)
converter = tf.lite.TFLiteConverter.from_saved_model('lstm_test')
converter.target_spec.supported_ops = [
        tf.lite.OpsSet.TFLITE_BUILTINS, # enable TensorFlow Lite ops.
        ]

tflite_model = converter.convert()

inpt = tf.random.normal([1,10,192])
init_state_h = tf.zeros([1,192])
init_state_c = tf.zeros([1,192])
init_state_h_tfl = tf.zeros([1,192])
init_state_c_tfl = tf.zeros([1,192])

#set up tflite interpreter
interpreter = tf.lite.Interpreter(model_content = tflite_model)
interpreter.allocate_tensors()
input_details = interpreter.get_input_details()
output_details = interpreter.get_output_details()
print('input_details')
print(input_details)
print('output details')
print(output_details)

for k in range(10):
    outpt, st_h, st_c = my_model.predict([tf.expand_dims(inpt[:,k,:], axis=1), init_state_h, init_state_c])
    init_state_h = st_h
    init_state_c = st_c
    interpreter.set_tensor(input_details[0]['index'], tf.expand_dims(inpt[:,k,:], axis = 1))
    interpreter.set_tensor(input_details[1]['index'], init_state_h_tfl)
    interpreter.set_tensor(input_details[2]['index'], init_state_c_tfl)
    interpreter.invoke()
    outpt_tfl = interpreter.get_tensor(output_details[0]['index'])
    st_h_tfl = interpreter.get_tensor(output_details[1]['index'])
    st_c_tfl = interpreter.get_tensor(output_details[2]['index'])
    init_state_h_tfl = st_h_tfl
    init_state_c_tfl = st_c_tfl
    np.testing.assert_almost_equal(outpt, outpt_tfl, decimal = 5)
```

### 3. Failure after conversion
When the above code is ran, it can be seen that there are considerable differences between the tflite model and the original tensorflow model.

**Comment by DerryFitz** [comment]
The dimension mismatch is probably related to another open issue: https://github.com/tensorflow/tensorflow/issues/58844
It happened once or twice to me while I was preparing the code for this bug report

**Comment by DerryFitz** [comment]
Also, I can replicate the bug back as far as tf 2.6, it seems to be ok in tf 2.5 if that is of any use

**Comment by DerryFitz** [comment]
Any update on this issue? it would really be helpful to have it resolved.

**Comment by ecanoc** [comment]
I'm experiencing the exact same issue. Any updates?

**Comment by DerryFitz** [comment]
any updates on resolving this issue? Thanks

**Comment by pkgoogle** [comment]
Hello @DerryFitz and @ecanoc,

The tflite conversion process does not guarantee the model's weights stay exactly the same even without any optimizations. One example is operators being fused. Obviously they shouldn't change too much, but it isn't guaranteed to stay exactly the same. I calculated the mse of the two outputs in this [gist](https://colab.research.google.com/gist/pkgoogle/59c4e3b6ce428321fcac7848009e3bb7/58858.ipynb). The mse's are quite small and thus I don't believe this is a bug.

Let us know if you are satisfied by this solution.

**Comment by DerryFitz** [comment]
Hi, I have ran the gist, and the mse is in the range 0.001-0.005. While this might be acceptable for some applications, when you are dealing with audio signals, an error in that range will be quite audible, and in my case leads to a noticeable deterioration in the results obtained from my full network. The error would need to be around 10^-6 or 10^-7 for it to be unnoticeable. As such, it means that converting a model containing an lstm to a tflite model is unuseable for audio since tf2.5. If tf is to remain viable for use in professional audio applications, this will need to be looked at.

**Comment by pkgoogle** [comment]
Thanks for the reply @DerryFitz,

Would a direct conversion with no performance improvements or optimizations essentially resolve your issue? For now I will label this a feature request rather than a bug for such a thing.

**Comment by DerryFitz** [comment]
Thanks for getting back to me, 
yes, being able to carry out a direct conversion on the lstmcell would essentially resolve the issue. It would be nice to have an optimised conversion which had the desired level of accuracy, but if the fastest way to resolve it in the short term is direct conversion, that would be sufficient.

**Comment by pkgoogle** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The TFLite team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the TFLite space.

Thanks.
